src/main/java/com/sngular/kloadgen/sampler/schemaregistry/impl/ApicurioSchemaRegistry.java

```diff
@@ -0,0 +1,149 @@
+package com.sngular.kloadgen.sampler.schemaregistry.impl;
+
+import io.apicurio.registry.resolver.SchemaParser;
+import io.apicurio.registry.rest.client.RegistryClient;
+import io.apicurio.registry.rest.client.RegistryClientFactory;
+import io.apicurio.registry.rest.client.exception.RestClientException;
+import io.apicurio.registry.rest.v2.beans.ArtifactMetaData;
+import io.apicurio.registry.rest.v2.beans.SearchedArtifact;
+import io.apicurio.registry.serde.SerdeConfig;
+import io.apicurio.registry.serde.avro.AvroSchemaParser;
+import io.apicurio.registry.serde.jsonschema.JsonSchemaParser;
+import io.apicurio.registry.serde.protobuf.ProtobufSchemaParser;
+import com.sngular.kloadgen.common.SchemaTypeEnum;
+import com.sngular.kloadgen.exception.KLoadGenException;
+import com.sngular.kloadgen.sampler.schemaregistry.schema.ApicurioParsedSchema;
+import com.sngular.kloadgen.sampler.schemaregistry.SchemaRegistryConstants;
+import com.sngular.kloadgen.sampler.schemaregistry.SchemaRegistryManager;
+import com.sngular.kloadgen.util.SchemaRegistryKeyHelper;
+import org.apache.tika.io.IOUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+public class ApicurioSchemaRegistry implements SchemaRegistryManager {
+
+    private RegistryClient schemaRegistryClient;
+    private Map<String, String> propertiesMap;
+
+    public ApicurioSchemaRegistry() {
+        this.propertiesMap = new HashMap<>();
+        this.propertiesMap.put(SchemaRegistryKeyHelper.SCHEMA_REGISTRY_NAME, SchemaRegistryConstants.SCHEMA_REGISTRY_APICURIO);
+        this.propertiesMap.put(SchemaRegistryKeyHelper.SCHEMA_REGISTRY_URL_KEY, SerdeConfig.REGISTRY_URL);
+    }
+
+    @Override
+    public String getSchemaRegistryUrlKey() {
+        return SerdeConfig.REGISTRY_URL;
+        // todo: si al final usamos el map deproperties tambien se podria devolver lo siguiente
```

Comment:
Remove this todo comment or if needed write it in English and remove the commented code line